src/extension.ts

```diff
@@ -486,6 +486,207 @@ export function activate(context: vscode.ExtensionContext) {
         }
     }));
 
+    function getAnimationFromJsonPointer(glTF, jsonPointer : string): { json: any, path: string } {
+        let inAnimation = false;
+        let inSampler = false;
+        const jsonPointerSplit = jsonPointer.split('/');
+        const numPointerSegments = Math.min(5, jsonPointerSplit.length);
+        let result = glTF;
+        let path = '';
+        const firstValidIndex = 1; // Because the path has a leading slash.
+        for (let i = firstValidIndex; i < numPointerSegments; ++i) {
+            inAnimation = inAnimation || (jsonPointerSplit[i] == 'animations');
+            inSampler = inAnimation && (inSampler || (jsonPointerSplit[i] == 'samplers'));
+            result = result[jsonPointerSplit[i]];
+            path += '/' + jsonPointerSplit[i];
+        }
+        if (!inSampler) {
+            vscode.window.showErrorMessage('Please select an animation sampler to import.');
+            result = undefined;
+        }
+        return { json: result, path: path };
+    }
+
+    //
+    // Import an animation for editing.
+    //
+    context.subscriptions.push(vscode.commands.registerTextEditorCommand('gltf.importAnimation', async (te, t) => {
+        if (!checkValidEditor()) {
+            return;
+        }
+
+        const map = tryGetJsonMap();
+        if (!map) {
+            return;
+        }
+
+        const bestKey = tryGetCurrentJsonPointer(map);
+        if (!bestKey) {
+            return;
+        }
+        const glTF = map.data;
+
+        const activeTextEditor = vscode.window.activeTextEditor;
+        const animationPointer = getAnimationFromJsonPointer(glTF, bestKey);
+        if (!animationPointer.json) {
+            return;
+        }
+
+        animationPointer.json.extras = animationPointer.json.extras || {};
+        for (const key of ['input', 'output']) {
+            const accessorId = animationPointer.json[key];
+            const accessor = glTF.accessors[accessorId];
+            const bufferView = glTF.bufferViews[accessor.bufferView];
+            const buffer = getBuffer(glTF, bufferView.buffer, activeTextEditor.document.fileName);
+            let accessorValues = getAccessorArrayBuffer(buffer, accessor, bufferView);
+            animationPointer.json.extras[`vscode_gltf_${key}`] = Array.from(accessorValues);
+            animationPointer.json.extras['vscode_gltf_type'] = accessor.type;
+        }
+
+        const pointer = map.pointers[animationPointer.path];
+
+        const newJson = JSON.stringify(animationPointer.json);
+        await activeTextEditor.edit(editBuilder => {
+            editBuilder.replace(new vscode.Range(pointer.value.line, pointer.value.column,
+                pointer.valueEnd.line, pointer.valueEnd.column), newJson);
+        });
+        await vscode.commands.executeCommand('editor.action.formatDocument', activeTextEditor.document.uri);
```

Comment:
Do we need to reformat the whole doc here?  Granted it will reformat on export, so I guess it's OK...

Reply:
No I'm changing this to use the indent settings as suggested above and indenting the edited selection 4 times.